main.tf

```
@@ -103,9 +103,31 @@ resource "aws_s3_object" "s3_dummy" {
 }
 
 resource "aws_lambda_function_event_invoke_config" "default" {
-  count                  = var.retries != null ? 1 : 0
+  count                  = var.retries != null || var.destination_on_failure != null || var.destination_on_success != null ? 1 : 0
   function_name          = aws_lambda_function.default.function_name
   maximum_retry_attempts = var.retries
+
+  dynamic "destination_config" {
+    for_each = var.destination_on_failure != null || var.destination_on_success != null ? 1 : 0
```

```suggestion
    for_each = for_each = var.destination_on_failure != null || var.destination_on_success != null ? { create : true } : {}
```

You need something to iterate over in a `for_each`, so we use this hack to create the dynamic block.